File: packages/cli/templates/stacks/mcp-go/internal/tools/echo.go
// Package tools holds pure tool implementations, testable without the SDK.
package tools

import "errors"

// ErrEmpty is returned when echo is given an empty string.
var ErrEmpty = errors.New("text must be a non-empty string")

// Echo returns its input verbatim.
func Echo(text string) (string, error) {
	if text == "" {
		return "", ErrEmpty
	}
	return text, nil
}

File: packages/cli/templates/stacks/mcp-python/src/tools/echo.py
"""Echo tool implementation — testable as a plain function."""


async def echo(text: str) -> str:
    """Returns its input verbatim.

    Raises:
        ValueError: when text is empty or None.
    """
    if not isinstance(text, str) or not text:
        raise ValueError("text must be a non-empty string")
    return text
